File: project-euler/Problem_06/problem_6.erl
-module(problem_6).
-compile(export_all).

start()->
    io:format("result=~p~n", [sum_of_n(100) * sum_of_n(100) - sum_of_square_n(100)]).

sum_of_n(Num)->
    lists:foldr(fun(X, Y)->
			X + Y
		end,
		0, lists:seq(1, Num)).

sum_of_square_n(Num)->
    lists:foldr(fun(X, Y) ->
			X + Y
		end,
		0, lists:map(fun(X)->
				     X * X
			     end,
			     lists:seq(1, Num))).


    
    

File: project-euler/Problem_06/problem_6.js
function sum_of_squares(num) {
    var result = 0;

    for (var i = 1; i <= num; i++) {
	result += i * i;
    }

    return result;
}

function square_of_sum(num) {
    var result = 0;

    for (var i = 1; i <= num; i++) {
	result += i;
    }

    return result * result;
}

function problem_6() {
    return square_of_sum(100) - sum_of_squares(100);
}

document.write('<b>result: </b>' + problem_6());